firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -4548,6 +4548,57 @@ private static FirebaseApp appForDatabaseUrl(String url, String name) {
         name);
   }
 
+  @Test
+  public void testGetPropagatesEventsToListeners() // TODO(mtewani): Convert into smoke test
+      throws DatabaseException, InterruptedException, ExecutionException {
+    /**
+     * There's a situation where if you have a pending get() and then an onValue() at the same
+     * location and get resolves before onValue does, onValue gets no listeners
+     */
+    int counter = 0;
+    boolean pendingGet = false;
+    while (counter++ < 5 || !pendingGet) {
+      FirebaseApp writeApp =
+          appForDatabaseUrl(IntegrationTestValues.getDatabaseUrl(), UUID.randomUUID().toString());
```

Comment:
It seems like all we really need in test is the `FirebaseDatabase`. Perhaps we should add a helper:

```
FirebaseDatabase readerDb = IntegrationTestHelpers.newFirebaseDatabase();
```

Just a wrapper over what you have here to make this a bit easier to read.

Reply:
Done.